eduNEXT/xblock-filesmanager
# feat: chonky backend integration

## Description

Chonky is a versatile library that empowers us to manage a hierarchical structure of folders and files. In this pull request (PR), the following features are introduced:

- **Upload files:** Seamlessly add new files to the tree structure.
- **Create folders:** Easily organize your content by creating new folders.
- **Delete files:** Remove unwanted files from the structure.
- **Delete folders:** Efficiently manage your content by deleting entire folders.
- **Download files:** Download individual files from the structure, with the ability to download one file at a time


![filesmanager-demo](https://github.com/eduNEXT/xblock-filesmanager/assets/134975835/19f49ee0-6a17-4c2a-92ad-e25dae6bb70d)


### How to test it

- Check this branch`jv/feat-chonky-integration-backend` in the XBlock files manager
- Create a new branch 

 ```
git checkout -b branch-to-test-integration
```
- Pull the changes made in [this pr](https://github.com/eduNEXT/xblock-filesmanager/pull/14)

 ```
git pull origin MJG/sync-directories
```
1. use node 18
 ```
nvm use
```

2. install dependencies
```
yarn install 
```
3 run to see any change in the react code 
```
yarn dev 
```

4. Click in edit Xblock
5. Click on "Upload files" button
6. Click on "save" button



**Merge checklist:**
Check off if complete *or* not applicable:
